app/main.py

```diff
@@ -1,2 +1,90 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length: int = 0
+        self.hash_table: list = [None] * 8
+
+    def hash_index(self, key: Hashable) -> int:
+        return hash(key) % len(self.hash_table)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self.hash_index(key)
+        if self.hash_table[index] is None:
+            self.hash_table[index]: list = []
+        for i, (k, v) in enumerate(self.hash_table[index]):
+            if k == key:
+                self.hash_table[index][i] = (key, value)
+                break
+        else:
+            self.hash_table[index].append((key, value))
+            self.length += 1
+            if self.length > len(self.hash_table) * 2 / 3:
+                self._resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self.hash_index(key)
+        if self.hash_table[index] is None:
+            raise KeyError("Key not found")
+        for k, v in self.hash_table[index]:
+            if k == key:
+                return v
+        raise KeyError("Key not found")
```

Comment:
Pretty much the same error, can you avoid such code duplication in this particular method?